### PR TITLE
[INLONG-2377] DataProxy support PB compression protocol format source.

### LINKED ISSUE #2377: [Feature] DataProxy support new protocol format source.
### Description

DataProxy support new protocol format source.
Parent issue: #2375

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luchunliang**
### Title Name: [INLONG-2377][Inlong-DataProxy] DataProxy support PB compression protocol format source.

Fixes #2377 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpChannelPipelineFactory.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    /**
+     * get message handlers
+     * 
+     * @param  cp
+     * @return
+     */
+    @SuppressWarnings("unchecked")
+    public ChannelPipeline addMessageHandlersTo(ChannelPipeline cp) {
+        cp.addLast("messageDecoder", new LengthFieldBasedFrameDecoder(
+                sourceContext.getMaxMsgLength(), 0, 4, -4, 0, true));
```
Comment (imvan) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpChannelPipelineFactory.java:76]: a series magic number

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+                Constructor ctor = clazz.getConstructor(SourceContext.class);
+
+                LOG.info("Using channel processor:{}", getChannelProcessor().getClass().getName());
+                pipelineFactory = (ChannelPipelineFactory) ctor.newInstance(sourceContext);
+                if (pipelineFactory instanceof Configurable) {
+                    this.pipelineFactoryConfigurable = ((Configurable) pipelineFactory);
+                    this.pipelineFactoryConfigurable.configure(sourceContext.getParentContext());
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
```
Comment (imvan) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java:124]: do not print exception directly

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+            try {
+                serverBootstrap.releaseExternalResources();
+            } catch (Exception e) {
+                LOG.warn("Inlong TCP Source serverBootstrap stop ex ", e);
+            } finally {
+                serverBootstrap = null;
+            }
+        }
+
+//        super.stop();
```
Comment (imvan) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java:182]: please remove useless code

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+            if (receiveBufferSize > 16 * 1024 * 1024) {
+                receiveBufferSize = 16 * 1024 * 1024;
+            }
+            Preconditions.checkArgument(receiveBufferSize > 0, "receiveBufferSize must be > 0");
+
+            sendBufferSize = context.getInteger(ConfigConstants.SEND_BUFFER_SIZE, 1024 * 64);
+            if (sendBufferSize > 16 * 1024 * 1024) {
+                sendBufferSize = 16 * 1024 * 1024;
+            }
+            Preconditions.checkArgument(sendBufferSize > 0, "sendBufferSize must be > 0");
```
Comment (imvan) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java:212]: 202-206 and 208-212 can be abstracted to a function

**Review by imvan — APPROVED** [review]

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java (hunk 2 above)
Comment (luchunliang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java:124]: fixed it.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java (hunk 3 above)
Comment (luchunliang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java:182]: fixed it.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java (hunk 4 above)
Comment (luchunliang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java:212]: fixed it.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+                    InlongTcpChannelHandler.class.getName());
+            messageHandlerName = messageHandlerName.trim();
+            Preconditions.checkArgument(StringUtils.isNotBlank(messageHandlerName),
+                    "messageHandlerName is empty");
+
+            if (this.pipelineFactoryConfigurable != null) {
+                this.pipelineFactoryConfigurable.configure(context);
+            }
+        } catch (Throwable t) {
+            t.printStackTrace();
```
[the remark this answers is not in the document]
Comment (luchunliang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java:234]: fixed it.

**Review by baomingyu — APPROVED** [review]
+1

**Pull request merged**